### PR TITLE
Don't log raw binary

### PR DESCRIPTION
**Pull request by anacrolix**
Causes issues particularly when logging to the terminal.

### CODE REVIEW

File: listen.go [hunk 1]
```diff
@@ -4,7 +4,6 @@ import (
 	"net"
 
 	pb "github.com/libp2p/go-libp2p-circuit/pb"
-
```
Comment (vyzo) [inline listen.go]: don't remove this line!

File: listen.go [hunk 2]
```diff
@@ -33,7 +32,7 @@ func (l *RelayListener) Accept() (manet.Conn, error) {
 		}
 
 		// TODO: Pretty print.
-		log.Infof("accepted relay connection: %s", c)
+		log.Infof("accepted relay connection: %q", c)
```
Comment (vyzo) [inline listen.go]: is this the right format specifier?
godoc says:
```
%q	a single-quoted character literal safely escaped with Go syntax.
```

Comment (anacrolix) [reply listen.go]: It's the best I've found in this kind of circumstance (unknown strings).

Comment (vyzo) [reply listen.go]: ok, let's go with it.

**Pull request merged**